streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordCollectorImpl.java

```diff
@@ -107,7 +114,9 @@ public void onCompletion(final RecordMetadata metadata, final Exception exceptio
                             final TopicPartition tp = new TopicPartition(metadata.topic(), metadata.partition());
                             offsets.put(tp, metadata.offset());
                         } else {
-                            if (sendException == null) {
+                            ProductionExceptionHandlerResponse exceptionAction = productionExceptionHandler.handle(serializedRecord, exception);
```

Comment:
Looks like this call can be conditionally executed when sendException == null

Reply:
That would effectively mean that the exception only occurs on the first production exception that occurs.... Hmmm. Yes, I think that's the more sensible decision because otherwise a handler could end up in a situation where it returns `CONTINUE` but the app still falls over because of a prior situation where it returned `FAIL`.

Good catch, I'll fix this.